src/test/java/spoon/test/filters/FilterTest.java

```diff
@@ -115,9 +115,17 @@ public void testNameFilter() throws Exception {
 		CtClass<?> foo = factory.Package().get("spoon.test.filters.testclasses").getType("Foo");
 		assertEquals("Foo", foo.getSimpleName());
 		List<CtNamedElement> elements = foo.getElements(new NameFilter<>("i"));
+		// contract: tests NameFilter constructor for cases other than null
 		assertEquals(1, elements.size());
+		// contract: tests NameFilter constructor for null value case
+		assertThrows(IllegalArgumentException.class, () -> foo.getElements(new NameFilter<>(null)));
+		// contract: tests the getType method of the NameFilter class
+		assertEquals("CtNamedElement",new NameFilter<>("i").getType().getSimpleName());
```

Comment:
Avoid unnecessary chains. Here, what you're checking is the return value of `getType`, so check that!

```suggestion
		assertEquals(CtNamedElement.class,new NameFilter<>("i").getType());
```

Reply:
will change this